editor/components/document-outline/index.js

```diff
@@ -95,9 +95,16 @@ export const DocumentOutline = ( { blocks = [], title, onSelect, isTitleSupporte
 	// Select the corresponding block in the main editor
 	// when clicking on a heading item from the list.
 	const onSelectHeading = ( uid ) => onSelect( uid );
+	const titleNode = document.querySelector( '.editor-post-title__input' );
+	const titleId = () => {
+		if ( typeof titleNode.getAttribute( 'id' ) ) {
+			titleNode.setAttribute( 'id', 'id-' + titleNode.innerHTML );
```

Comment:
We are changing a dom node rendered react directly, that may have undesirable and unexpected effects.
Would keep the same events we had before and just use an anchor instead of a button be an option?